src/Catalyst.Core.Modules.Ledger/Web3EthApi.cs

```diff
@@ -22,25 +22,52 @@
 #endregion
 
 using System;
+using Autofac.Features.AttributeFilters;
+using Catalyst.Abstractions.Consensus.Deltas;
 using Catalyst.Abstractions.Kvm;
 using Catalyst.Abstractions.Ledger;
+using Catalyst.Abstractions.Ledger.Models;
+using Catalyst.Core.Modules.Ledger.Repository;
+using Nethermind.Core.Crypto;
+using Nethermind.Evm;
 using Nethermind.Store;
 
 namespace Catalyst.Core.Modules.Ledger 
 {
-    public class Web3EthApi : IWeb3EthApi
+    public class Web3EthApi : IWeb3EthApi, ITransactionReceiptResolver
     {
-        public Web3EthApi(IDeltaExecutor deltaExecutor, IStateReader stateReader, IStateRootResolver stateRootResolver, IDeltaResolver deltaResolver)
+        private readonly ITransactionReceiptRepository _receipts;
+        public const string ComponentName = nameof(Web3EthApi);
+            
+        public Web3EthApi(IStateReader stateReader, IDeltaResolver deltaResolver, IDeltaCache deltaCache, [KeyFilter(ComponentName)] ITransactionProcessor processor, [KeyFilter(ComponentName)] IStorageProvider storageProvider, [KeyFilter(ComponentName)] IStateProvider stateProvider, ITransactionReceiptRepository receipts)
         {
-            DeltaExecutor = deltaExecutor ?? throw new ArgumentNullException(nameof(deltaExecutor));
+            _receipts = receipts;
             StateReader = stateReader ?? throw new ArgumentNullException(nameof(stateReader));
-            StateRootResolver = stateRootResolver ?? throw new ArgumentNullException(nameof(stateRootResolver));
             DeltaResolver = deltaResolver ?? throw new ArgumentNullException(nameof(deltaResolver));
+            DeltaCache = deltaCache ?? throw new ArgumentNullException(nameof(deltaCache));
+            Processor = processor ?? throw new ArgumentNullException(nameof(processor));
+            StorageProvider = storageProvider ?? throw new ArgumentNullException(nameof(storageProvider));
+            StateProvider = stateProvider ?? throw new ArgumentNullException(nameof(stateProvider));
         }
         
-        public IDeltaExecutor DeltaExecutor { get; }
         public IStateReader StateReader { get; }
         public IDeltaResolver DeltaResolver { get; }
-        public IStateRootResolver StateRootResolver { get; }
+        public IDeltaCache DeltaCache { get; }
+        
+        public object SyncRoot => Processor;
+        public ITransactionProcessor Processor { get; }
+        public IStorageProvider StorageProvider { get; }
+        public IStateProvider StateProvider { get; }
+        public ITransactionReceiptResolver ReceiptResolver => this;
+
+        public TransactionReceipt Find(Keccak hash)
+        {
+            if (_receipts.TryFind(hash, out TransactionReceipt receipt))
+            {
+                return receipt;
+            }
+
+            throw new Exception("Receipt not found");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)